chore: improve `Ecosystem.md` linter to lint all sections

nooreldeensalah opened the pull request:
Extend the ecosystem linter functionality (module name validation + ordering) to all `Ecosystem.md` sections (`Core`, `Community`, `Community Tools`) instead of `Community` only.

Relevant Issues/PRs: #4256 #4257

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/fastify/fastify/blob/master/CONTRIBUTING.md

By making a contribution to this project, I certify that:

* (a) The contribution was created in whole or in part by me and I
  have the right to submit it under the open source license
  indicated in the file; or

* (b) The contribution is based upon previous work that, to the best
  of my knowledge, is covered under an appropriate open source
  license and I have the right under that license to submit that
  work with modifications, whether created in whole or in part
  by me, under the same open source license (unless I am
  permitted to submit under a different license), as indicated
  in the file; or

* (c) The contribution was provided directly to me by some other
  person who certified (a), (b) or (c) and I have not modified
  it.

* (d) I understand and agree that this project and the contribution
  are public and that a record of the contribution (including all
  personal information I submit with it, including my sign-off) is
  maintained indefinitely and may be redistributed consistent with
  this project or the open source license(s) involved.
-->

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

mcollina reviewed (approved):
lgtm

jsumners commented on .github/scripts/lint-ecosystem.js:
```diff
@@ -19,15 +19,29 @@ module.exports = async function ({ core }) {
   for await (const line of rl) {
     lineNumber += 1
+
+    if (line.startsWith('- [') === true) {
```
We can cache this `startsWith` since it is used again later.

nooreldeensalah replied:
Thank you for your suggestion, I've added a commit to address this change.
I've also added two more commits:
- One commit to extend the functionality of the ordering to all sections.
- Another commit to fix out of order modules in the `Core` section.
If you feel like the last two commits are irrelevant to this PR, or not required, I can drop them if you wish

jsumners reviewed (approved):
Looks good to me.

nooreldeensalah commented:
I've added a few commits to extend the functionality to all sections (ordering + validation), and I've fixed a few out of order modules in the `Core` section

Eomm commented:
Could you merge `main`?
CI is red because of #4260
